src/Groups/perm.jl

```diff
@@ -772,15 +779,36 @@ end
 
 ################################################################################
 #
-#   perm
+#   @perm
 #
+
 @doc raw"""
-    @perm ex
-    
-Input a permutation in cycle notation. Supports arbitrary expressions for
-generating the integer entries of the cycles. The parent group is inferred 
-to be the symmetric group with a degree of the highest integer referenced 
-in the permutation.
+    @perm expr
+    @perm n expr
+    @perm G expr
+
+Input a permutation or a non-empty list of permutations in cycle notation.
+
+Supports arbitrary expressions for generating the integer entries of the cycles.
+
+`expr` may either be a single permutation, a non-empty list of permutations,
+or a non-empty tuple of permutations. **Note:** `@perm ()` denotes the identity
+permutation, NOT the empty tuple of permutations.
+
+If a group `G` is provided, the permutations are created as elements of `G`. This will
+raise an error if the permutations are not elements of `G`.
+
+If an integer `n` is provided, the permutations are created as elements of the
+symmetric group of degree `n`, i.e., `symmetric_group(n)`.
+
+Input a list of permutations in cycle notation, created as elements of the
+symmetric group of degree `n`, i.e., `symmetric_group(n)`, by invoking
+[`cperm`](@ref) suitably.
+
+In the remaining case, the parent group is inferred to be the symmetric group
+with a degree of the highest integer referenced in `expr`. This may result in
+evalutating the expressions in the cycle entries multiple times, so it is
```

Comment:
```suggestion
evaluating the expressions in the cycle entries multiple times, so it is
```
OK for now, but we could avoid that, couldn't we? By storing the result of the evaluation immediately as vectors (of vectors), and then later e.g. passing those to `cperm`

Reply:
We can only avoid it for the case of a single permutation, and indeed the previous code did. I removed that for the time being to reduce Tha number of case distinctions.
For the vector and tuple cases, we cannot let the cperm dispatch handle the detection. Consider e.g. `@perm [(1, 2), (3,4)]`. if we would delegate that to `cperm([1, 2])` and `cperm([3,4])`, then the parents would differ (`Sym(2)` vs. `Sym(4)`.

I have some ideas on how to avoid the double evaluation, but I need to think a bit more about it and test if it regressed performance for the case of just ints in all entries. Thus, I would like to move that to a follow-up PR